### PR TITLE
[android] prefetch gradle dependencies to docker image

### PR DESCRIPTION
**Pull request by IvanKobzarev**
As a fix for gradle build instability https://github.com/pytorch/pytorch/issues/29159

### CODE REVIEW

File: common/install_android.sh [hunk 1]
```diff
@@ -104,3 +104,83 @@ export PATH="${GRADLE_HOME}/bin/:${PATH}"
 echo "PATH:${PATH}"
 
 gradle --version
+
+rm -rf /tmp/gradledeps
```
Comment (dreiss) [inline common/install_android.sh]: redundant?

File: common/install_android.sh [hunk 2]
```diff
@@ -104,3 +104,83 @@ export PATH="${GRADLE_HOME}/bin/:${PATH}"
...
+
+_gradledeps=/tmp/gradledeps
+
+rm -rf $_gradledeps
```
Comment (dreiss) [inline common/install_android.sh]: Please put quotes around the variable: `$_gradledeps`

File: common/install_android.sh [hunk 3]
```diff
@@ -104,3 +104,83 @@ export PATH="${GRADLE_HOME}/bin/:${PATH}"
...
+
+cat <<EOF >"$_gradledeps/AndroidManifest.xml"
+<manifest package="org.pytorch" />
+EOF
+
+cat <<EOF >"$_gradledeps/build.gradle"
```
Comment (dreiss) [inline common/install_android.sh]: Can this be a separate checked-in file instead of a heredoc?

File: common/install_android.sh [hunk 4]
```diff
@@ -104,3 +104,83 @@ export PATH="${GRADLE_HOME}/bin/:${PATH}"
...
+    }
+
+    sourceSets {
+        main {
+            manifest.srcFile 'AndroidManifest.xml'
+        }
+    }
+}
+
+dependencies {
```
Comment (dreiss) [inline common/install_android.sh]: Can we add fbjni-java-only to this?

File: common/install_android.sh [hunk 5]
```diff
@@ -104,3 +104,83 @@ export PATH="${GRADLE_HOME}/bin/:${PATH}"
 echo "PATH:${PATH}"
 
 gradle --version
+
+rm -rf /tmp/gradledeps
+
+_gradledeps=/tmp/gradledeps
```
Comment (dreiss) [inline common/install_android.sh]: Why put this in /tmp?

**Review by dreiss — APPROVED** [review]

File: common/install_android.sh (hunk 3 above)
Comment (IvanKobzarev) [reply common/install_android.sh]: It's possible if change /ubuntu/Dockerfile that has:
```
ADD ./common/install_android.sh install_android.sh
RUN if [ -n "${ANDROID}" ]; then bash ./install_android.sh; fi
RUN rm install_android.sh
```
and the same for other docker images.

```
ADD ./android/build.gradle build.gradle
ADD ./android/AndroidManifest.xml AndroidManifest.xml
...
```
That will be a bit different with other images, but in separate files

File: common/install_android.sh (hunk 4 above)
Comment (IvanKobzarev) [reply common/install_android.sh]: I tried with only fbjni-java-only - offline build does not work due to no deps in cache for:
'com.google.code.findbugs:jsr305:3.0.1'
'com.facebook.soloader:nativeloader:0.8.0'

if to add them explicitly - works.

I will try to understand why this happens

File: common/install_android.sh (hunk 5 above)
Comment (IvanKobzarev) [reply common/install_android.sh]: I planned to remove it eventually, as deps should be stored in `~/.gradle/caches`

**Comment by ezyang** [comment]
Let me know when you want to merge this

**Pull request closed without merging**